From 9bc3cedba54708c40c4a853b240c46e69f87de3c Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 17 Mar 2020 10:04:40 -0700
Subject: [PATCH] Makefile: Add LIBDIR variable

This ensures that it can be installed into custom location and also

Upstream-Status: Submitted
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/Makefile
+++ b/Makefile
@@ -1,5 +1,5 @@
 ARCH := $(shell uname -m)
-
+LIBDIR := /lib
 CFLAGS = -ggdb3 -O2 -Wall -Iarch/${ARCH}
 
 LIBUCONTEXT_C_SRC = $(wildcard arch/${ARCH}/*.c)
@@ -10,8 +10,8 @@ LIBUCONTEXT_SOVERSION = 0
 LIBUCONTEXT_NAME = libucontext.so
 LIBUCONTEXT_STATIC_NAME = libucontext.a
 LIBUCONTEXT_SONAME = libucontext.so.${LIBUCONTEXT_SOVERSION}
-LIBUCONTEXT_PATH = /lib/${LIBUCONTEXT_SONAME}
-LIBUCONTEXT_STATIC_PATH = /lib/${LIBUCONTEXT_STATIC_NAME}
+LIBUCONTEXT_PATH = ${LIBDIR}/${LIBUCONTEXT_SONAME}
+LIBUCONTEXT_STATIC_PATH = ${LIBDIR}/${LIBUCONTEXT_STATIC_NAME}
 
 all: ${LIBUCONTEXT_SONAME} ${LIBUCONTEXT_STATIC_NAME}
 
@@ -36,9 +36,9 @@ clean:
 		${LIBUCONTEXT_OBJ} test_libucontext
 
 install: all
-	install -D -m755 ${LIBUCONTEXT_NAME} ${DESTDIR}/${LIBUCONTEXT_PATH}
-	install -D -m664 ${LIBUCONTEXT_STATIC_NAME} ${DESTDIR}/${LIBUCONTEXT_STATIC_PATH}
-	ln -sf ${LIBUCONTEXT_SONAME} ${DESTDIR}/lib/${LIBUCONTEXT_NAME}
+	install -D -m755 ${LIBUCONTEXT_NAME} ${DESTDIR}${LIBUCONTEXT_PATH}
+	install -D -m664 ${LIBUCONTEXT_STATIC_NAME} ${DESTDIR}${LIBUCONTEXT_STATIC_PATH}
+	ln -sf ${LIBUCONTEXT_SONAME} ${DESTDIR}${LIBDIR}/${LIBUCONTEXT_NAME}
 
 check: test_libucontext ${LIBUCONTEXT_SONAME}
 	env LD_LIBRARY_PATH=$(shell pwd) ./test_libucontext
